### ISSUE TITLE
Kubelet v1.10.11 zeroes out all extended resources upon restart, causing all pods to be evicted

### ISSUE THREAD

**Issue by PaulFurtado** [kubernetes/kubernetes#73948]
**What happened**:
Upon kubelet restart, it zeroes out every extended resource on the node, which caused every pod using extended resources on the node to get evicted. We use extended resources with all of our pods, and during a routine kubelet restart, it wiped out a good chunk of the cluster before we noticed what was happening. We caught this in QA, so it wasn't the end of the world. 

**What you expected to happen**:
Kubelet to not zero out all of the extended resources on restart.

**How to reproduce it (as minimally and precisely as possible)**:
1. Advertise an extended resource from a node
2. Create a pod that requests the resource
3. Restart kubelet
4. Watch pod get evicted

**Anything else we need to know?**:
- The issue was almost definitely caused by this PR: https://github.com/kubernetes/kubernetes/pull/66881
- It looks like the original issue it was intended to fix is: https://github.com/kubernetes/kubernetes/issues/64632

I understand the motivation of that change, but I really wonder if this was the intended consequence. It also seems like there's no way you can restart kubelet when using extended resources because it will zero out the resources for some period of time before you can possibly patch the node capacity back to the correct state, making extended resources effectively unusable. It sounds like we might be able to work around this by writing a device plugin, but we really didn't want to implement a daemon on the critical path to deal with advertising basic and unchanging resources which the nodes offer. Maybe, we could also just specify the resources in the kubelet config file? That way kubelet could use the values from the config file on restart rather than resetting to zero.

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"archive", BuildDate:"2018-12-04T17:57:19Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: ec2 and gcp clusters
- OS (e.g. from /etc/os-release): CentOS 6.9
- Kernel (e.g. `uname -a`): `4.14.77-hs623.el6.x86_64` (internal build)
- Install tools: (internal/puppet)
- Others: N/A

**Comment by Pingan2017** [comment]
same issue !
Maybe we can add a whitelist of extended resources to avoid this problem.
/cc @yujuhong @derekwaynecarr @kubernetes/sig-node-bugs
If this idea can be accepted, I am glad to work on this.

**Comment by PaulFurtado** [comment]
@pingan2017 check out the discussion on https://github.com/kubernetes/kubernetes/issues/64632

**Comment by PaulFurtado** [comment]
/remove-lifecycle stale

I'm also a bit concerned that no progress has been made on such a destructive bug...

**Comment by chardch** [comment]
Hi @PaulFurtado,

Is it the case that this issue never wants the extended resources to be zero'd out, or just not upond kubelet restart? Also to confirm, you are not using a device plugin?

**Comment by calixwu** [comment]
Same issue.. Any progress?

**Comment by chardch** [comment]
@calixwu This should be handled by #77699

**Comment by PaulFurtado** [comment]
> Is it the case that this issue never wants the extended resources to be zero'd out, or just not upond kubelet restart? Also to confirm, you are not using a device plugin?

@chardch Sorry, I missed that comment. But yeah, I expect resources to not be zeroed out upon kubelet restart. If I deleted the node object and recreated it, that seems like a perfectly reasonable time to zero out resources unless that happens during normal cluster upgrade operations.

We are not using a device plugin for these basic integer resources; we use the integer resources to fill in basic kubernetes accounting gaps like how many pod IPs are available on different nodes. Writing and running a grpc server to serve up a static map of integers would be pretty crazy. That said, I think the ideal way to expose the configuration of these integer resources would be via the kubelet config, and then there'd be no ambiguity on kubelet restart.

Anyway, there are two elements to this issue:
1. Ending up with this feature exposed in a different way
2. Fixing an extremely serious regression in a documented API that used to work fine. The GKE team seems to only have cared about their use case here (uninstalling device plugins?) and for everyone else using the feature, we just got 100% of our pods immediately hard evicted upon kubelet restart after a minor version upgrade of kubelet. It's clear that other people were also burned by this issue.
